packages/device_info/lib/device_info.dart

```diff
@@ -0,0 +1,231 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
```

Comment:
> This plugin is independent of the crash reporting use-case.

I agree Yegor. My point is that if you take crash reporting out of the picture, there's not much else left as to why you would need this info on Dart side.

> Could you please point to an example for how to send a custom crash report?

For Android, they have [this](https://firebase.google.com/docs/crash/android#create_your_first_error) which is "crash" reporting even if the app doesn't crash. They don't support this for iOS but you can still do [this](https://firebase.google.com/docs/crash/ios#create_custom_logs). So sending a simple message/log from Dart side onFlutterError and reporting this to FirebaseCrash is possible. The downside is that it will have to actually crash your app. That might be what you expect to do anyway https://github.com/flutter/flutter/issues/10695. 

My overall point is that whenever you are doing some sort of reporting out of user's device (feedback, crash reporting, collecting logs etc), it is better to use native libraries. Not only you get the benefit of all this information collected, you also get additional benefits such libraries provide such as offline mode, batching/compressing logs etc. We could implement all that ourselves but... why? :)